search/tasks_test.py

```diff
@@ -40,3 +54,121 @@ def test_increment_document_integer_field_task(mocked_api):
     increment_document_integer_field(*indexing_api_args)
     assert mocked_api.increment_document_integer_field.call_count == 1
     assert mocked_api.increment_document_integer_field.call_args[0] == indexing_api_args
+
+
+def test_wrap_retry_exception():
+    """wrap_retry_exception should raise RetryException when other exceptions are raised"""
+    with assert_not_raises():
+        with wrap_retry_exception(KeyError):
+            # Should not raise an exception
+            pass
+
+
+@pytest.mark.parametrize("matching", [True, False])
+def test_wrap_retry_exception_matching(matching):
+    """A matching exception should raise a RetryException"""
+
+    class SubError(KeyError):
+        """Use a subclass to assert isinstance use"""
+
+    def raise_thing():
+        """raise the exception"""
+        if matching:
+            raise SubError()
+        else:
+            raise TabError()
+
+    matching_exception = RetryException if matching else TabError
+    with pytest.raises(matching_exception):
+        with wrap_retry_exception(KeyError):
+            raise_thing()
+
+
+def test_index_post_with_comments(mocker):
+    """index_post should call the api function of the same name"""
+    index_post_mock = mocker.patch('search.indexing_api.index_post_with_comments')
+    wrap_mock = mocker.patch('search.tasks.wrap_retry_exception')
```

Comment:
This only shows up in 2 test cases (this one, `test_index_channel`), but I think it's worth it to get rid of some boilerplate by turning this patch and our assertions into a fixture:

```python
@pytest.fixture()
def wrap_retry_mock():
    """Patches the wrap_retry_exception context manager and asserts that it was called by any test that uses it"""
    wrap_mock = mocker.patch('search.tasks.wrap_retry_exception')
    yield
    wrap_mock.assert_called_once_with(PrawcoreException, PRAWException)

def test_index_post_with_comments(mocker, wrap_retry_mock):
    # ...
```